File: demo/src/main/scala/com/huawei/dli/DataPreparation.scala
package com.huawei.dli

import org.apache.spark.sql.SparkSession

object DataPreparation {
  def main(args: Array[String]): Unit = {

    if (args.length != 2) {
      println(s"Wrong number of arguments! Expected 2 but got ${args.length}")
      return
    }

    // your AK/SK to access OBS
    val AK = "******"
    val SK = "******"

    val prefix = "s3a://" + AK + ":" + SK + "@"
    val readPath = prefix + args(0)
    val writePath = prefix + args(1)

    val spark = SparkSession
      .builder
      .appName("demo")
      .getOrCreate()

    // read the raw data
    val data = spark.read.textFile(readPath)

    // you can do some transformations like filter, map, etc on data
    val cleanedData = data.filter { line =>
      // if the type of the first attribute is not Long, remove this record
      var isValid = true
      try {
        val attributes = line.split(",")
        attributes.head.toLong
      } catch {
        case _: Throwable =>
          isValid = false
      }
      isValid
    }

    // output the cleaned data we want
    cleanedData.write.text(writePath)

    spark.close()
  }
}

File: demo/src/main/python/data_preparation.py
import sys

from pyspark.sql import SparkSession

# check if the type of the first attribute is not Long
def headIsLong(x):
  arr = x.split(',')
  isValid = True
  try:
    longV = int(arr[0])
  except ValueError:
    isValid = False
  return isValid

if __name__ == "__main__":
  if len(sys.argv) != 2:
    print "Wrong number of arguments! Expected 2 but got " + str(len(sys.argv))
    exit(-1)

  # your AK/SK to access OBS
  ak = "******"
  sk = "******"

  prefix = "s3a://" + ak + ":" + sk + "@"
  readPath = prefix + sys.argv[1]
  writePath = prefix + sys.argv[2]

  spark = SparkSession.builder.appName("demo").getOrCreate()

  # read the raw data
  data = spark.read.text(readPath).rdd.map(lambda r: r[0])

  # if the type of the first attribute is not Long, remove this record
  cleanedData = data.filter(headIsLong)

  # output the cleaned data we want
  cleanedData.saveAsTextFile(writePath)

  spark.stop()
